flyteplugins/go/tasks/pluginmachinery/internal/webapi/core.go

```diff
@@ -65,25 +68,66 @@ func (c CorePlugin) GetProperties() core.PluginProperties {
 	return core.PluginProperties{}
 }
 
-func (c CorePlugin) Handle(ctx context.Context, tCtx core.TaskExecutionContext) (core.Transition, error) {
+func (c CorePlugin) toSyncPlugin() (webapi.SyncPlugin, error) {
+	plugin, ok := c.p.(webapi.SyncPlugin)
+	if !ok {
+		return nil, fmt.Errorf("Core plugin does not implement the sync plugin interface")
+	}
+	return plugin, nil
+}
+
+func (c CorePlugin) toAsyncPlugin() (webapi.AsyncPlugin, error) {
+	plugin, ok := c.p.(webapi.AsyncPlugin)
+	if !ok {
+		return nil, fmt.Errorf("Core plugin does not implement the async plugin interface")
+	}
+	return plugin, nil
+}
+
+func (c CorePlugin) syncHandle(ctx context.Context, tCtx core.TaskExecutionContext) (core.Transition, error) {
+	plugin, err := c.toSyncPlugin()
+	if err != nil {
+		return core.UnknownTransition, err
```

Comment:
Does it make sense to fail the task/workflow execution immediately? I don't know how `core.UnknownTransition` drives the state machine.